### PR TITLE
[react-native] Upgrade to 0.62.2

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Let's have latest React Native in the upcoming SDK!

# How

See the story at https://www.notion.so/expo/React-Native-Upgrade-Diary-947bc0b506a942189fd47ff6e53bf95b.

# Test Plan

Expo Client runs, versioned SDK38 also runs (after a couple of post-version fixes outlined in the aforementioned doc).

`expo-cli` can't handle `client_log` events, but that's something we may want to figure out after merging this PR, I guess.

### CODE REVIEW

**Comment by sjchmiela** [comment]
Some of the commits may be left not-reviewed as they are effect of automated processes. Some changes worth reviewing are:
- commits applied on top of `0.62-stable` branch (@ide)
- `LogBoxModule` exception suppression (@ide), [story](https://www.notion.so/expo/React-Native-Upgrade-Diary-947bc0b506a942189fd47ff6e53bf95b#a8e5c632736940709f0be306645e6b3d), [fix](https://github.com/expo/react-native/commit/8666ac5826e593627cd371385e649dbd85b91569)
- decision not to version JNI (@ide) [story](https://www.notion.so/expo/React-Native-Upgrade-Diary-947bc0b506a942189fd47ff6e53bf95b#5e98076b273448af85bd31222a9ac288)
- `symbolicateStackTrace` fix (@ide, @fson) [commit](https://github.com/expo/expo/pull/8310/commits/85f73ab3ebc554cbd7b7677dc48d2bf2bbebe119)
- Gradle upgrade across the repo (@mczernek) [commit](https://github.com/expo/expo/pull/8310/commits/7bb610e6c93f1f1c3710e175115d64fca8877ce1)
- added `uiMode` to Android Activities (@bbarthec) [commit](https://github.com/expo/expo/pull/8310/commits/469a9cf9de65161d3f51e8bc95885b450622b757)
- `soloader` upgrade from 0.6.0 to 0.8.2 (0.8.0 used in RN 0.62, I saw reports of problems on 0.8.0, eg. [this](https://github.com/facebook/react-native/issues/25923#issuecomment-625823694)) (@ide) [commit](https://github.com/expo/expo/pull/8310/commits/f3bdfc63507856a5e23e0f3d5bed03077a6e3e0b)
- removed custom resolution of `@react-native-community/cli-platform-ios` (@IjzerenHein) [commit](https://github.com/expo/expo/pull/8310/commits/58332ad73181d0269a220c3dc6eba1179d0b232d), pods installed ok 🤷‍♂️ 
- fixed `expotools` for versioning of RN 0.62 (@tsapeta) [commit](https://github.com/expo/expo/pull/8310/commits/b8a0d964c9fa3d5193abae33dbdb26dafd3c98a8)

**Comment by ide** [comment]
For the react-native branch, to keep things organized we have several `expo/***` branches each with one commit. If there are new commits we need for sdk-38, or commits that were on the sdk-37 branch that do not have a corresponding `expo/***` branch, please create an `expo/***` branch for each one that is rebased either on top of upstream/master@HEAD or the most recent ancestor commit between upstream/master and 0.62-stable. Thank you for being thorough when going through the list of commits. The more we can upstream (or split into code that is upstreamable so that Expo-specific functionality can just be in the Expo code without needing a patch) the better, too.

The LogBox fix seems OK for now, but reaffirms that replacing the various devtools and devsupport classes will likely make the platform more stable.

Not versioning JNI seems simplest for now. I do want to be careful, though, because needing to patch several modules is not sustainable.

The symbolicateStackTrace change looks fine to me. I left some small inline comments. Make sure the unit tests continue to pass or update them if needed.

Removing the custom Yarn package resolution is fantastic. Thank you.

File: template-files/android/AndroidManifest.xml [hunk 1]
```diff
@@ -98,15 +98,15 @@
 
     <activity
       android:name=".experience.ExperienceActivity"
-      android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
+      android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
```
Comment (bbarthec) [inline template-files/android/AndroidManifest.xml:101]: Ok, so this change will have to be inspected from the perspective of `react-native-appearance` 🤔 
I'm not sure whether `react-native-appearance` can handle it properly (change from `recreating whole activity` to `fire some lifecycle method upon change`)? cc @brentvatne

File: package.json [hunk 2]
```diff
@@ -29,8 +29,5 @@
     "eslint": "^6.8.0",
     "expo-yarn-workspaces": "*",
     "prettier": "^1.19.0"
-  },
-  "resolutions": {
-    "@react-native-community/cli-platform-ios": "3.0.0"
```
Comment (bbarthec) [inline package.json:34]: It's probably safe. As far as I know `@r-n-c/cli-platform-ios` there not this much that this change could break.

**Review by tsapeta — COMMENTED** [review]
Changes in expotools and tools look good to me 💪. Thanks for taking care of this and testing out how versioning works 🤗

File: packages/expo/src/logs/LogSerialization.ts [hunk 3]
```diff
@@ -117,7 +117,9 @@ async function _symbolicateErrorAsync(error: Error): Promise<StackFrame[]> {
   const parsedStack = parseErrorStack(error);
   let symbolicatedStack: StackFrame[] | null;
   try {
-    symbolicatedStack = await symbolicateStackTrace(parsedStack);
+    // @ts-ignore: symbolicateStackTrace has different real/Flow declaration
+    // than the one in DefinietelyTyped.
```
[the remark this answers is not in the document]
Comment (sjchmiela) [reply packages/expo/src/logs/LogSerialization.ts:121]: This is one of the few words I think I know in English and I just can't type it right. 😅

**Comment by sjchmiela** [comment]
`sdk` CI fails due to not-changed changelog in `expo-gl`. The only change made there was upgrading `react-test-renderer` in `devDependencies` which is not a user-facing change, so it doesn't need to end up in CHANGELOG.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**